### PR TITLE
Search contexts: Support context keyword in the search frontend

### LINKED ISSUE #16363: Support context: keyword in the search frontend
Includes:
- Highlighting
- Autocompletion
- Hovers

### PR DESCRIPTION
**Pull request by novoselrok**
Adds support for context filter to the frontend. Includes syntax highlighted `@` prefix.

### CODE REVIEW

**Review by rvantonder — APPROVED** [review]

File: client/shared/src/search/query/filters.ts [hunk 1]
```diff
@@ -136,6 +136,10 @@ export const FILTERS: Record<NegatableFilter, NegatableFilterDefinition> &
         singular: true,
         suggestions: 'RepoGroup',
     },
+    [FilterType.context]: {
+        description: 'Search only repositories within a specified context',
```
Comment (limitedmage) [inline client/shared/src/search/query/filters.ts:140]: @quinnkeast can you double-check this hint string? (I think it's OK)

**Review by limitedmage — APPROVED** [review]

File: client/shared/src/search/query/filters.ts (hunk 1 above)
Comment (novoselrok) [reply client/shared/src/search/query/filters.ts:140]: I checked with him beforehand and this was his suggestion.

Comment (limitedmage) [reply client/shared/src/search/query/filters.ts:140]: Awesome, thanks for verifying this!

**Pull request merged**